sunpy/timeseries/sources/goes.py

```diff
@@ -222,50 +223,52 @@ def _parse_netcdf(filepath):
 
         Parameters
         ----------
-        filepath : `~str`
+        filepath : `str`
             The path of the file to parse
         """
-        with h5netcdf.File(filepath, mode="r", **XRSTimeSeries._netcdf_read_kw) as d:
-
-            header = MetaDict(OrderedDict(d.attrs))
-            if "a_flux" in d.variables:
-                xrsa = np.array(d["a_flux"])
-                xrsb = np.array(d["b_flux"])
-                xrsa_quality = np.array(d['a_flags'])
-                xrsb_quality = np.array(d['b_flags'])
-
-                start_time_str = d["time"].attrs["units"]
-                if not isinstance(start_time_str, str):
-                    # For h5netcdf<0.14
-                    start_time_str = start_time_str.astype(str)
-                start_time_str = start_time_str.lstrip("seconds since").rstrip("UTC")
-                # Perform the time addition in UTime format to ignore leap seconds
-                times = Time(parse_time(start_time_str).utime + d["time"], format="utime")
-            elif "xrsa_flux" in d.variables:
-                xrsa = np.array(d["xrsa_flux"])
-                xrsb = np.array(d["xrsb_flux"])
-                xrsa_quality = np.array(d['xrsa_flags'])
-                xrsb_quality = np.array(d['xrsb_flags'])
-
-                start_time_str = d["time"].attrs["units"]
-                if not isinstance(start_time_str, str):
-                    # For h5netcdf<0.14
-                    start_time_str = start_time_str.astype(str)
-                start_time_str = start_time_str.lstrip("seconds since")
-                # Perform the time addition in UTime format to ignore leap seconds
-                times = Time(parse_time(start_time_str).utime + d["time"], format="utime")
-
-            else:
-                raise ValueError(f"The file {filepath} doesn't seem to be a GOES netcdf file.")
+        with h5netcdf.File(filepath, mode="r", **XRSTimeSeries._netcdf_read_kw) as h5nc:
+            header = MetaDict(OrderedDict(h5nc.attrs))
+            flux_name = h5nc.variables.get("a_flux") or h5nc.variables.get("xrsa_flux")
+            if flux_name is None:
+                raise ValueError(f"No flux data (either a_flux or xrsa_flux) found in file: {filepath}")
+            flux_name = flux_name.name
+            xrsa = np.array(h5nc[flux_name])
+            xrsb = np.array(h5nc[flux_name.replace("a", "b")])
```

Comment:
introduce a second variable for xrsb instead of the replacement, same for the flags as it makes it difficult to read the code

Reply:
I have created one for xrsb.